### PR TITLE
Fix for BOWC-1092

### PR DESCRIPTION
**Pull request by sirdeggen**
Associate Paymail with User ID - /api/alias allows blank body and just creates a paymail "" to point to a new user ID with no profile.

Expected :  /api/alias should not allow registering empty paymail aliases nor allow profiles without at minimum and email and name. Error 400 otherwise.

### CODE REVIEW

File: data/payd/payd.go [hunk 1]
```diff
@@ -117,6 +117,9 @@ func (p *Payd) baseURL() string {
 
 func (p *Payd) CreateUser(ctx context.Context, req models.UserDetails) (*models.User, error) {
 	var user *models.User
+	if req.Name == "" || req.Email == "" {
+		return nil, errors.New("Must include name and email for user registration.")
```
Comment (Tigh-Gherr) [inline data/payd/payd.go:121]: Can start with lower case as per the rationale here https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2151

File: service/alias.go [hunk 2]
```diff
@@ -39,19 +42,15 @@ func (svc *alias) CreateAlias(ctx context.Context, a *models.NewAliasDetails) *m
 		PhoneNumber: a.PhoneNumber,
 	})
 	if err != nil {
-		return &models.AliasResponse{
-			Error: errors.Wrap(err, "User creation failed"),
-		}
+		return nil, errors.Wrap(err, "User creation failed")
```
Comment (Tigh-Gherr) [inline service/alias.go:45]: error casing

File: service/alias.go [hunk 3]
```diff
@@ -39,19 +42,15 @@ func (svc *alias) CreateAlias(ctx context.Context, a *models.NewAliasDetails) *m
...
 	newAlias := &models.AliasResponse{
 		UserID:  user.ID,
 		Paymail: a.Paymail,
 	}
 	err = svc.str.CreateAlias(ctx, newAlias)
 	if err != nil {
-		return &models.AliasResponse{
-			Error: errors.Wrap(err, "Alias creation failed"),
-		}
+		return nil, errors.Wrap(err, "Alias creation failed")
```
Comment (Tigh-Gherr) [inline service/alias.go:53]: error casing

**Review by Tigh-Gherr — CHANGES_REQUESTED** [review]
Just some error casing to sort, will approve once done

**Comment by sirdeggen** [comment]
> Just some error casing to sort, will approve once done

@Tigh-Gherr please may I request another glance - fixes done and another capability added that's actually potentially useful this time - rather than a dummy.

**Review by Tigh-Gherr — APPROVED** [review]
LGTM!

**Pull request merged**